hnn_core/gui/gui.py

```diff
@@ -26,15 +26,307 @@
                              _read_legacy_params)
 from hnn_core.viz import plot_dipole
 
-THEMECOLOR = "#8A2BE2"
 
+class HNNGUI:
+    """HNN GUI class
+
+    Parameters
+    ----------
+    theme_color : str
+        The theme color of the whole dashboard.
+    log_window_height: str
+        The height of the log output window (in pixel).
+    visualization_window_width: str
+        The width of the visualization window (in pixel).
+    visualization_window_height: str
+        The height of the visualization window (in pixel).
+    left_sidebar_width: str
+        The width of the left side bar (in pixel).
+    TODO: add more parameters.
+
+    Attributes
+    ----------
+    params: dict
+        The parameters to use for constructing the network.
+    TODO: add more attributes
+    """
+
+    def __init__(self, theme_color="#8A2BE2", log_window_height="100px",
+                 visualization_window_width="1000px",
+                 visualization_window_height="500px",
+                 left_sidebar_width='380px',
+                 drive_widget_width="200px"):
+        # set up styling.
+        self.theme_color = theme_color
+        self.log_window_height = log_window_height
+        self.visualization_window_width = visualization_window_width
+        self.visualization_window_height = visualization_window_height
+        self.left_sidebar_width = left_sidebar_width
+        self.drive_widget_width = drive_widget_width
+
+        # load default parameters
+        self.params = self.load_parameters()
+
+        # In-memory storage of all simulation related variables
+        self.variables = dict(net=None, dpls=list())
+
+        # Simulation parameters
+        self.tstop = FloatText(value=170,
```

Comment:
I think you should have `self.widgets` which is a dictionary and similarly `self.callbacks` which is a dictionary. The dictionaries have matching key names ... that way it will be easy to inspect the object

Reply:
`self.layouts` is a good point, but I'd like to keep all widgets and callbacks as direct member of the HNNGUI class, which makes documentation and tests easier

Reply:
The object should not have > 10 attributes ... otherwise it will be a mess